src/compile/layer.ts

```diff
@@ -100,6 +101,17 @@ export class LayerModel extends Model {
     return this.children.reduce((db, child) => child.assembleSelectionData(db), data);
   }
 
+  public assembleGroupStyle(): string | string[] {
+    const uniqueStyles = new Set<string>();
+    for (const child of this.children) {
+      for (const style of array(child.assembleGroupStyle())) {
+        uniqueStyles.add(style);
+      }
+    }
+    const styles = Array.from(uniqueStyles);
```

Comment:
I think you can check the length of the set and return the first item of the set instead. No need to make it an array.

Reply:
To get the first item you need to get the iterator and call next, which seems equally expensive as this but result in a more complicated code. So I think this is actually better.

Reply:
I disagree. The array call makes a copy while creating an iterator is highly optimized.

Reply:
I guess since we may return an array, it doesn't make a big difference.

Reply:
Most of the time N=1 or a few.  Maybe iterator is marginally better, but produces way longer code that doesn't justify the benefit.